src/app.py

```diff
@@ -194,6 +197,87 @@ def render_form(**kwargs):
                        new_url=new_url)
 
 
+@app.route("/pull_request_hook", methods=["POST"])
+def pull_request_hook():
+    event = request.headers.get('X-GitHub-Event')
+    if event == "ping":
+        return Response('Pong')
+    elif event != "pull_request":
+        return Response("Wrong event type", status=400)
+
+    payload = request.get_json()
+    action = payload['action']
+    if action not in ("opened", "edited", "reopened", "synchronize"):
+        return Response("Not Interested")
+    # status=204 : No Content
+
+    sender     = payload['sender']['login']
+
+    head_sha   = payload['pull_request']['head']['sha']
+    head_label = payload['pull_request']['head']['label']
+    head_ref   = payload['pull_request']['head']['ref']
+
+    base_sha   = payload['pull_request']['base']['sha']
+    base_label = payload['pull_request']['base']['label']
+    base_ref   = payload['pull_request']['base']['ref']
+
+    pull_num   = payload['pull_request']['number']
+    pull_url   = payload['pull_request']['html_url']
+
+    pull_ref   = "pull/{pull_num}/head".format(**locals())
+
+    # make sure data repo contains relevant commits
+    stdout, stderr, ret = fetch_data_ref(base_ref, pull_ref)
+
+    if ret == 0:
+        script = src_dir + "/tests/automerge_downtime_ok.py"
+        cmd = [script, base_sha, head_sha, sender]
+        stdout, stderr, ret = runcmd(cmd, cwd=global_data.topology_data_dir)
+
+    OK = "Yes" if ret == 0 else "No"
+
+    subject = "Pull Request {pull_url} {action}".format(**locals())
+
+    out = """\
+In Pull Request: {pull_url}
+GitHub User '{sender}' wants to merge branch {head_label}
+        (at commit {head_sha})
+into {base_label}
+        (at commit {base_sha})
+
+Eligible for downtime automerge? {OK}
+
+automerge_downtime script output:
+---
+{stdout}
+---
+{stderr}
+---
+""".format(**locals())
+
+    recipients = [ x + "@cs.wisc.edu" for x in ("edquist", "matyas", "blin") ]
+    _,_,_ = send_mailx_email(subject, out, recipients)
+
+    return Response(out)
+
+
+def runcmd(cmd, input=None, **kw):
+    if input is None:
+        stdin = None
+    else:
+        stdin = PIPE
+    p = subprocess.Popen(cmd, stdout=PIPE, stderr=PIPE, stdin=stdin,
+                         encoding='utf-8', **kw)
+    stdout, stderr = p.communicate(input)
+    return stdout, stderr, p.returncode
+
+def fetch_data_ref(*refs):
+    return runcmd(['git', 'fetch', 'origin'] + list(refs),
+                  cwd=global_data.topology_data_dir)
+
+def send_mailx_email(subject, body, recipients):
+    return runcmd(["mailx", "-s", subject] + recipients, input=body)
```

Comment:
Can you make this run in the background instead? You're not getting the results so don't block the webapp.

Reply:
I've never seen `mailx` block on the topology hosts (for more than .03s), even if sending to an invalid domain ... certainly the call takes much less time to complete than the other parts of this webhook.  So i'm not really convinced that running this in the background (not waiting on the result) will be of any benefit.  Especially if we will soon be posting a PR comment rather than sending an email.

But um, despite my feedback, do you still want me to do it anyway?

Reply:
I think you don't need to do this if you do as @bbockelm recommends and put the webhook in a separate webapp.